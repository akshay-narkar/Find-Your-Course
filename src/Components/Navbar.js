/*eslint-disable*/
import { Link } from 'react-router-dom';
import Navloggedin from './Navbarloggedin';
import Navloggedout from './Navbarloggedout';

export default function Navbar() {
  return (
 
    <div>
      <nav className="navbar navbar-expand-lg navbar-light bg-light">
        <div className="container-fluid">
          <Link to = {{pathname: '/'}} className = "navbar-brand" aria-current = "page">Upskill</Link>
          <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
            <span className="navbar-toggler-icon" />
          </button>
          <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <ul className="navbar-nav me-auto mb-2 mb-lg-0">
              
              <li className="nav-item">
               <Link to = {{pathname: '/'}} className = "nav-link active" aria-current = "page">Home</Link>
              </li>
              {sessionStorage.getItem("uid") ? <Navloggedin /> : <Navloggedout />  }
            {/* <Link to={{ pathname: '/error' }} className="link">
              <li className="nav-item">
                <a className="nav-link" href="#">Link</a>
              </li>
              </Link> */}
              {/* <li className="nav-item dropdown">
                <a className="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
                  Dropdown
                </a>
                <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
                  <li><a className="dropdown-item" href="#">Action</a></li>
                  <li><a className="dropdown-item" href="#">Another action</a></li>
                  <li><hr className="dropdown-divider" /></li>
                  <li><a className="dropdown-item" href="#">Something else here</a></li>
                </ul>
              </li> */}
    
            </ul>
          </div>
        </div>
      </nav>
    </div>
  );
}
