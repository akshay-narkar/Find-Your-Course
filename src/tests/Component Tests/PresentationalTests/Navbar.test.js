import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../../../Components/Navbar';

//   const error = 'Random error'
//   const clickHandler1 = () => {};

it('UI is unaltered wrt first snapshot', () => {
  const elem = renderer.create(<BrowserRouter><Navbar /></BrowserRouter>).toJSON();

  expect(elem).toMatchSnapshot();
});
