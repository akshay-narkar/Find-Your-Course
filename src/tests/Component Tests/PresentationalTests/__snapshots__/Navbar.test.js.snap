// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI is unaltered wrt first snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand fw-bold"
        href="/"
        onClick={[Function]}
      >
        Upskill
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active fw-light"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              className="nav-link active fw-light"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active fw-light"
              href="/signup"
              onClick={[Function]}
            >
              SignUp
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
